### PR TITLE
AirfRANS: OOD task baselines (scarce + reynolds with AdamW)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Round 1 only tested the `full` AirfRANS task. The benchmark also defines `scarce` (low-data regime) and `reynolds` (OOD Reynolds number) tasks. These test generalization in ways the `full` task cannot — and paper comparisons require reporting on the full task family. This PR establishes baseline numbers on scarce and reynolds, using both the default optimizer and the Round 1 winner (AdamW lr=5e-4).

## Instructions

Run 4 parallel trials on 4 GPUs. Use slices=64 throughout.

```bash
cd target/icml2026

# v0: full task + AdamW lr=5e-4 + slices=64 (control vs baseline #2423)
CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset airfrans --airfrans_task full \
  --optimizer adamw --lr 5e-4 --model_slices 64 \
  --wandb_group airfrans-ood-tasks \
  --wandb_run_name v0-full-adamw &

# v1: scarce task + default optimizer (Lion lr=3e-4)
CUDA_VISIBLE_DEVICES=1 python train.py \
  --dataset airfrans --airfrans_task scarce \
  --optimizer lion --lr 3e-4 --model_slices 64 \
  --wandb_group airfrans-ood-tasks \
  --wandb_run_name v1-scarce-lion &

# v2: scarce task + AdamW lr=5e-4
CUDA_VISIBLE_DEVICES=2 python train.py \
  --dataset airfrans --airfrans_task scarce \
  --optimizer adamw --lr 5e-4 --model_slices 64 \
  --wandb_group airfrans-ood-tasks \
  --wandb_run_name v2-scarce-adamw &

# v3: reynolds task + AdamW lr=5e-4
CUDA_VISIBLE_DEVICES=3 python train.py \
  --dataset airfrans --airfrans_task reynolds \
  --optimizer adamw --lr 5e-4 --model_slices 64 \
  --wandb_group airfrans-ood-tasks \
  --wandb_run_name v3-reynolds-adamw &

wait
```

Report `val_primary/surface_mse` for each variant and task. For scarce, also note how many training samples are used. Note: `scarce` and `reynolds` tasks have different dataset sizes, so epoch timing may differ.

## Baseline

- **AirfRANS full task:** val_primary/surface_mse = 0.3308 (PR #2423, AdamW lr=5e-4)
- **AirfRANS scarce/reynolds:** No baseline yet — this PR establishes them.

## Results

_To be filled in by student_

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT norman:

## Results

### Summary Table

| Variant | Task | Optimizer | lr | Train Samples | Epochs | val_primary/surface_mse | test_primary/surface_mse | W&B run |
|---------|------|-----------|----|---------------|--------|------------------------|--------------------------|---------|
| v0 | full | AdamW | 5e-4 | 720 | 2 | 0.5201 | 0.5041 | jl4xixx3 |
| v1 | scarce | Lion | 3e-4 | 180 | 2 | 0.3646 | 0.8021 | cpdfxwn0 |
| v2 | scarce | AdamW | 5e-4 | 180 | 2 | 0.3351 | 0.8103 | 5u1t0qql |
| v3 | reynolds | AdamW | 5e-4 | 454 | 2 | 0.5956 | 0.8999 | tbhas6pq |

**Baseline comparison (full task):** val=0.3308, test=0.3199 (PR #2423, AdamW lr=5e-4, slices=96)

### Per-variant details

**v0 — full task + AdamW lr=5e-4 + slices=64 (control)**
- val_primary/surface_mse: epoch 1 = 0.6537, epoch 2 = 0.5201
- test surface_mse breakdown: Ux=0.0098, Uy=0.0006, p=2.0055, nut=0.0004
- test volume_mse: 0.8608
- Peak GPU memory: ~64 GB (GPU 0)

**v1 — scarce task + Lion lr=3e-4**
- val_primary/surface_mse: epoch 1 = 0.3646 (best), epoch 2 = 0.3917
- test surface_mse breakdown: Ux=0.0002, Uy=0.0015, p=3.2028, nut=0.0040
- test volume_mse: 1.0407
- Peak GPU memory: ~54 GB (GPU 1)

**v2 — scarce task + AdamW lr=5e-4**
- val_primary/surface_mse: epoch 1 = 0.3962, epoch 2 = 0.3351 (best)
- test surface_mse breakdown: Ux=0.0379, Uy=0.0274, p=3.1654, nut=0.0103
- test volume_mse: 1.0860
- Peak GPU memory: ~75 GB (GPU 2)

**v3 — reynolds task + AdamW lr=5e-4**
- val_primary/surface_mse: epoch 1 = 0.6307, epoch 2 = 0.5956 (best)
- test surface_mse breakdown: Ux=0.0018, Uy=0.0031, p=3.5895, nut=0.0053
- test volume_mse: 1.1333
- Peak GPU memory: ~52 GB (GPU 3)

### Commands used

```bash
# v0
CUDA_VISIBLE_DEVICES=0 python train.py --dataset airfrans --airfrans-task full --optimizer adamw --lr 5e-4 --model-slices 64 --agent norman --wandb-name "norman/v0-full-adamw" --wandb-group airfrans-ood-tasks

# v1
CUDA_VISIBLE_DEVICES=1 python train.py --dataset airfrans --airfrans-task scarce --optimizer lion --lr 3e-4 --model-slices 64 --agent norman --wandb-name "norman/v1-scarce-lion" --wandb-group airfrans-ood-tasks

# v2
CUDA_VISIBLE_DEVICES=2 python train.py --dataset airfrans --airfrans-task scarce --optimizer adamw --lr 5e-4 --model-slices 64 --agent norman --wandb-name "norman/v2-scarce-adamw" --wandb-group airfrans-ood-tasks

# v3
CUDA_VISIBLE_DEVICES=3 python train.py --dataset airfrans --airfrans-task reynolds --optimizer adamw --lr 5e-4 --model-slices 64 --agent norman --wandb-name "norman/v3-reynolds-adamw" --wandb-group airfrans-ood-tasks
```

### What happened

1. **v0 control (full/slices=64) vs baseline (full/slices=96):** val_primary/surface_mse = 0.5201 vs 0.3308 baseline — a significant regression. Reducing model slices from 96 to 64 clearly hurts performance on the full task. The slices=64 model has less representational capacity.

2. **Scarce task — AdamW beats Lion:** v2 (AdamW, val=0.3351) outperforms v1 (Lion, val=0.3646) by ~8%, consistent with AdamW's advantage seen on the full task in PR #2423. However, test metrics are very close (0.8021 vs 0.8103), suggesting the val gap may not generalize perfectly.

3. **Reynolds task is harder:** v3 shows val=0.5956 and test=0.8999, both higher than the full task, which is expected — the OOD Reynolds number split tests extrapolation to unseen flow regimes. Pressure dominates the error (p=3.59 test).

4. **All runs limited to 2 epochs** by the 30-minute timeout. Loss curves were still decreasing at cutoff — all tasks would likely benefit from longer training.

5. **Pressure dominates surface_mse across all tasks**: the p component accounts for 96-99%+ of total surface_mse. This pattern is consistent across full, scarce, and reynolds tasks.

### Suggested follow-ups

- Run scarce and reynolds with slices=96 (default) to establish proper baselines — slices=64 may be artificially handicapping these new tasks.
- Run with longer training budget (more epochs or higher timeout) to see if the scarce task converges faster due to its 4x smaller dataset.
- Consider task-specific hyperparameter tuning — scarce/reynolds may benefit from different lr schedules given their different dataset sizes and difficulty profiles.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2432 because ADVISOR: Closing — first OOD baselines established (scarce val=0.3351, reynolds val=0.5956) but all at slices=64 with only 2 epochs, confounded by epoch starvation. Key finding: massive val/test gap on OOD tasks (scarce: val~0.33 but test~0.80; reynolds: val~0.60 but test~0.90) — the OOD generalization challenge is very real and pressure dominates OOD error even more severely (test pressure component ~3.2x per-channel for scarce vs baseline 1.28). These are useful directional baselines. Follow-up: re-run best OOD configs with no-EMA, slices=96, single job per task to get proper epoch counts. Norman should continue with OOD + no-EMA in the next round..
